src/Compilers/CSharp/Portable/LanguageVersion.cs

```diff
@@ -184,19 +202,18 @@ public static bool TryParse(this string version, out LanguageVersion result)
                     result = LanguageVersion.Latest;
                     return true;
 
+                case "7.1":
+                    result = LanguageVersion.CSharp7_1;
+                    return true;
+
                 default:
-                    // We are likely to introduce minor version numbers after C# 7, thus breaking the
-                    // one-to-one correspondence between the integers and the corresponding
-                    // LanguageVersion enum values. But for compatibility we continue to accept any
-                    // integral value parsed by int.TryParse for its corresponding LanguageVersion enum
-                    // value for language version C# 6 and earlier (e.g. leading zeros are allowed)
-                    int versionNumber;
-                    if (int.TryParse(version, NumberStyles.None, CultureInfo.InvariantCulture, out versionNumber) &&
-                        versionNumber <= 6 &&
-                        ((LanguageVersion)versionNumber).IsValid())
+                    if (int.TryParse(version, NumberStyles.None, CultureInfo.InvariantCulture, out int versionNumber) && versionNumber <= 7)
```

Comment:
Do we now allow "07"? Consider reporting errors for leading zeros on any version number.